core/extension/gdextension_interface.cpp

```diff
@@ -700,6 +700,91 @@ static GDExtensionTypeFromVariantConstructorFunc gdextension_get_variant_to_type
 	ERR_FAIL_V_MSG(nullptr, "Getting Variant conversion function with invalid type");
 }
 
+static GDExtensionInterfaceVariantGetInternalPtrFunc gdextension_get_variant_get_internal_ptr_func(GDExtensionVariantType p_type) {
+	switch (p_type) {
+		case GDEXTENSION_VARIANT_TYPE_BOOL:
+			return reinterpret_cast<GDExtensionInterfaceVariantGetInternalPtrFunc>(static_cast<bool *(*)(Variant *)>(VariantInternal::get_bool));
```

Comment:
Do we need to have the two casts? Could this just be:

```suggestion
			return reinterpret_cast<GDExtensionInterfaceVariantGetInternalPtrFunc>(&VariantInternal::get_bool);
```

These lines are quite difficult to read with the function pointer types, and I'm not sure they gain us anything?

Reply:
> Do we need to have the two casts? 
To my own chagrin, yes.

The inner cast is needed to select the correct function among the two (const and non-const), and the outer is needed to cast to the return type, which is incompatible because of its return type (`void *` vs typed).